### PR TITLE
adding out as outdir and adding entrypoint for npm

### PR DESCRIPTION
**Pull request by softchris**
# Pull Request Template

## Description

- the idea is to build to `out` as outDir
- adding some ignore patterns so we don't get __mocks__ and __tests__ with us when building
- also adding `main` and `files` to make `npm pack` work properly and to add correct entry point

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How to Test

- Replace this with instructions on how to test this PR

### CODE REVIEW

**Comment by softchris** [comment]
questions:
- do we need source maps, we could exclude those or maybe they are good to have in the beta phase to help us debug issues?
- do we need to add files explicitly that are not part of the defaults, https://docs.npmjs.com/files/package.json#files
- should we have two build scripts `build:beta` with source maps and `build:latest` without?

**Comment by bnb** [comment]
This is what would be shipped to npm at present. If there's anything missing, it needs to be added to `files`.

```
npm notice === Tarball Contents === 
npm notice 2.1kB package.json                               
npm notice 124B  CHANGELOG.md                               
npm notice 1.1kB License.txt                                
npm notice 9.4kB README.md                                  
npm notice 458B  out/builders/actions/deploy.d.ts           
npm notice 6.7kB out/builders/actions/deploy.js             
npm notice 4.2kB out/builders/actions/deploy.js.map         
npm notice 320B  out/builders/deploy.builder.d.ts           
npm notice 2.6kB out/builders/deploy.builder.js             
npm notice 1.5kB out/builders/deploy.builder.js.map         
npm notice 112B  out/builders/logout.builder.d.ts           
npm notice 1.4kB out/builders/logout.builder.js             
npm notice 385B  out/builders/logout.builder.js.map         
npm notice 11B   out/ng-add/index_spec.d.ts                 
npm notice 981B  out/ng-add/index_spec.js                   
npm notice 581B  out/ng-add/index_spec.js.map               
npm notice 229B  out/ng-add/index.d.ts                      
npm notice 3.1kB out/ng-add/index.js                        
npm notice 1.4kB out/ng-add/index.js.map                    
npm notice 1.1kB out/util/azure/account.d.ts                
npm notice 8.0kB out/util/azure/account.js                  
npm notice 5.0kB out/util/azure/account.js.map              
npm notice 324B  out/util/azure/auth.d.ts                   
npm notice 2.9kB out/util/azure/auth.js                     
npm notice 1.4kB out/util/azure/auth.js.map                 
npm notice 355B  out/util/azure/locations.d.ts              
npm notice 2.5kB out/util/azure/locations.js                
npm notice 1.9kB out/util/azure/locations.js.map            
npm notice 624B  out/util/azure/resource-group-helper.d.ts  
npm notice 2.0kB out/util/azure/resource-group-helper.js    
npm notice 690B  out/util/azure/resource-group-helper.js.map
npm notice 479B  out/util/azure/resource-group.d.ts         
npm notice 4.7kB out/util/azure/resource-group.js           
npm notice 2.6kB out/util/azure/resource-group.js.map       
npm notice 254B  out/util/azure/subscription.d.ts           
npm notice 2.8kB out/util/azure/subscription.js             
npm notice 1.4kB out/util/azure/subscription.js.map         
npm notice 582B  out/util/prompt/list.d.ts                  
npm notice 3.4kB out/util/prompt/list.js                    
npm notice 2.2kB out/util/prompt/list.js.map                
npm notice 115B  out/util/prompt/name-generator.d.ts        
npm notice 1.5kB out/util/prompt/name-generator.js          
npm notice 488B  out/util/prompt/name-generator.js.map      
npm notice 169B  out/util/prompt/spinner.d.ts               
npm notice 2.1kB out/util/prompt/spinner.js                 
npm notice 965B  out/util/prompt/spinner.js.map             
npm notice 694B  out/util/shared/types.d.ts                 
npm notice 110B  out/util/shared/types.js                   
npm notice 120B  out/util/shared/types.js.map               
npm notice 906B  out/util/workspace/angular-json.d.ts       
npm notice 4.0kB out/util/workspace/angular-json.js         
npm notice 3.0kB out/util/workspace/angular-json.js.map     
npm notice 594B  out/util/workspace/azure-json.d.ts         
npm notice 2.0kB out/util/workspace/azure-json.js           
npm notice 1.6kB out/util/workspace/azure-json.js.map   
```

File: package.json [hunk 1]
```diff
@@ -1,6 +1,10 @@
 {
   "name": "@azure/ng-deploy",
   "version": "0.1.0-beta.3",
+  "main": "out/index.js",
```
Comment (johnpapa) [inline package.json]: I like having `index.js` as the entry point. However, after i run `npm run build` no file exists in the `/out` folder (just sub folders). I think we should have an entry point ... right?

File: tsconfig.json [hunk 2]
```diff
@@ -32,6 +33,8 @@
     "src/**/*"
   ],
   "exclude": [
-    "src/*/files/**/*"
+    "src/*/files/**/*",
```
Comment (johnpapa) [inline tsconfig.json]: what is this `src/*/files` folder?

**Comment by johnpapa** [comment]
@softchris - I also have 51 changed files now :) We should add the `out` folder to `.gitignore` as part of this so we dont commit every build

File: tsconfig.json (hunk 2 above)
Comment (softchris) [reply tsconfig.json]: it was there before, not sure why it flags this as a change.. I think it is a remnant from when there was a sandbox environment.. it's a @shmool question :)

**Comment by johnpapa** [comment]
@softchris - are you ready to merge this?

File: tsconfig.json (hunk 2 above)
Comment (shmool) [reply tsconfig.json]: Added comma in the end of the line...

Comment (shmool) [reply tsconfig.json]: The file was generated by the schematics CLI this way

File: package.json [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
+  "main": "out/ng-add/index.js",
+  "files": [
+    "out/",
+    "builders.json",
+    "collection.json"
+  ],
   "description": "@azure/ng-deploy - Deploy Angular apps to Azure using the Angular CLI",
   "scripts": {
-    "build": "tsc -p tsconfig.json",
+    "build": "tsc -p tsconfig.json && npm run copy:builders:json && npm run copy:ngadd:json",
```
Comment (shmool) [inline package.json]: `npm start` doesn't work as expected anymore. It should keep running in watch mode. Instead, the process finishes.

**Pull request merged**